addons/account/models/account_move.py

```diff
@@ -2314,7 +2314,10 @@ def action_switch_invoice_into_refund_credit_note(self):
                         'debit' : line_vals['credit'],
                         'credit' : line_vals['debit']
                     })
-            move.write({'invoice_line_ids' : [(5, 0, 0)]})
+            # We're transforming an invoice to a refund: were we wanted the partner's bank account
+            # before, we now wants the company's one
+            company_bank = move.company_id.partner_id.bank_ids[:1]
+            move.write({'invoice_line_ids' : [(5, 0, 0)], 'invoice_partner_bank_id': company_bank and company_bank.id})
```

Comment:
Perhaps we should just write `False` on the field `invoice_partner_bank_id` instead? I see no reason while selecting the first one we find is better than just removing the value altogether.

Reply:
I think the first one is fine but it's my opinion :)

Reply:
alright let's go with that then. It was from a bugfix, but I honestly don't really agree either. I prefer no data rather than arbitrary data